### PR TITLE
hoc2023: AI header

### PR DESCRIPTION
**Pull request by breville**
This change adds to the AI modal a header which closely resembles the styling of the AI block.

The goal is to help the user understand that what is shown inside the modal, including the purple blocks to set foreground and background, exists inside the single AI block.

The existing content stays mostly the same, though this offered a good chance to move the explanation text to the bottom.

### screenshot

Screenshot 2023-10-20 at 3 00 58 PM

### recording 

https://github.com/code-dot-org/code-dot-org/assets/2205926/676f8464-c068-49b8-92ec-1593d1e00227

### CODE REVIEW

File: apps/src/dance/ai/dance-ai-modal.module.scss [hunk 1]
```diff
@@ -299,5 +342,11 @@
         font-size: 16px;
       }
     }
+
+    .buttonsAreaTopRight {
```
Comment (bencodeorg) [inline apps/src/dance/ai/dance-ai-modal.module.scss]: This feels not great...acknowledging that all of the UI specifics are very in flux, but this feels like the type of thing that wouldn't be fixed as designs get updated? Is there a more semantic HTML container to put the button in that would make this feel less odd (looks like its not in the same container as the new block preview header, where I'd expect it)?

Same comment re: lots of !important being thrown in the `.dialog` styling, do we have any options there (or is the styling so far from done that it doesn't feel worthwhile to get right)?

File: apps/src/dance/ai/dance-ai-modal.module.scss [hunk 2]
```diff
@@ -61,30 +65,69 @@
...
+        margin-right: 6px;
+      }
+    }
+  }
 
   .innerArea {
-    width: 100%;
-    height: 100%;
+    width: calc(100% - 16px);
+    height: calc(100% - 50px - 16px - 8px);
```
Comment (sanchitmalhotra126) [inline apps/src/dance/ai/dance-ai-modal.module.scss]: Similar to Ben's comment, I know the UI is very much subject to change but this feels like a lot of magic numbers - can we create constants or drop a comment for where these numbers come from?

File: apps/src/dance/ai/dance-ai-modal.module.scss (hunk 1 above)
Comment (breville) [reply apps/src/dance/ai/dance-ai-modal.module.scss]: Good call.  I had hoped to keep all the buttons together, but for now have moved these two buttons to the header.

I'd love to get rid of the `!important` tags, but they'd require a change in how the shared `AccessibleDialog` component works.  There's an interesting internal thread about it [here](https://codedotorg.slack.com/archives/C03DBDN67B7/p1694201912817799).  I think that would be best done separately to this change.

File: apps/src/dance/ai/dance-ai-modal.module.scss (hunk 2 above)
Comment (breville) [reply apps/src/dance/ai/dance-ai-modal.module.scss]: Good call.  Updated with both constants and comments.

File: apps/src/dance/ai/dance-ai-modal.module.scss (hunk 1 above)
Comment (bencodeorg) [reply apps/src/dance/ai/dance-ai-modal.module.scss]: Thanks for the context re: `!important`. Moving the button into the header looks like a nice improvement to me.

**Review by bencodeorg — APPROVED** [review]
Looks really nice :)

**Review by sanchitmalhotra126 — APPROVED** [review]
Nice!

**Pull request merged**